src/api/hooks/useGetANS.ts

```diff
@@ -0,0 +1,65 @@
+import {useEffect, useState} from "react";
+import {NetworkName} from "../../constants";
+import {useGlobalState} from "../../GlobalState";
+
+function getFetchNameUrl(network: NetworkName, address: string) {
+  switch (network) {
+    case "mainnet":
+      return `https://www.aptosnames.com/api/mainnet/v1/name/${address}`;
+    case "testnet":
+      return `https://www.aptosnames.com/api/testnet/v1/name/${address}`;
+    default:
+      return undefined;
+  }
+}
+
+export function useGetNameFromAddress(address: string) {
+  const [state, _] = useGlobalState();
+  const [name, setName] = useState<string>();
```

Comment:
I guess that can be `string | null/undefined`...?

Reply:
It'll only be `string | undefined`, and with the default value unset the type is inferred as `string | undefined`. but let me add it.